src/compositemark/boxplot.ts

```diff
@@ -2,86 +2,66 @@ import {isNumber} from 'vega-util';
 import {Channel} from '../channel';
 import {Config} from '../config';
 import {reduce} from '../encoding';
+import {GenericMarkDef, isMarkDef, MarkConfig} from '../mark';
 import {AggregatedFieldDef, BinTransform, CalculateTransform, TimeUnitTransform} from '../transform';
+import {Flag, keys} from '../util';
 import {Encoding, forEach} from './../encoding';
 import {Field, FieldDef, isContinuous, isFieldDef, PositionFieldDef, vgField} from './../fielddef';
 import * as log from './../log';
-import {MarkConfig} from './../mark';
 import {GenericUnitSpec, LayerSpec} from './../spec';
 import {Orient} from './../vega.schema';
-import {getMarkSpecificConfigMixins} from './common';
+import {getMarkDefMixins} from './common';
 
 
-export const BOXPLOT: 'box-plot' = 'box-plot';
-export type BOXPLOT = typeof BOXPLOT;
-export type BoxPlotStyle = 'boxWhisker' | 'box' | 'boxMid';
+export const BOXPLOT: 'boxplot' = 'boxplot';
```

Comment:
Why is this changing to `boxplot`? I believe `box-plot` was a decision decision we came up with Jeff. Also box plot is two words and if we use `boxplot` then it'll be inconsistent with `error-bar`

Reply:
I want config to be easily referred.  `config["box-plot"]` is awkward.     

`config.boxplot/errorbar` are better.